kong-ngx-build

```diff
@@ -411,24 +411,28 @@ parse_version() {
   [[ -z $1 ]] && fatal 'missing arg $1 when invoking parse_version()'
   [[ -z $2 ]] && fatal 'missing arg $2 when invoking parse_version()'
 
+  local ver
   local subj=$1
-  local -n ver=$2
-  local re='^(-rc[0-9]+$)?[^0-9]*([0-9a-zA-Z]*)(.*?)$'
+  local re='^(-rc[0-9]+$)?[^0-9]*([0-9a-zA-Z]*)(.*)$'
 
   while [[ $subj =~ $re ]]; do
     if [[ ${BASH_REMATCH[1]} != "" ]]; then
-      ver+=( "${BASH_REMATCH[1]}" )
+      ver="$ver.${BASH_REMATCH[1]}"
     fi
 
     if [[ ${BASH_REMATCH[2]} != "" ]]; then
-      ver+=( "${BASH_REMATCH[2]}" )
+      ver="$ver.${BASH_REMATCH[2]}"
     fi
 
     subj="${BASH_REMATCH[3]}"
     if [[ $subj == "" ]]; then
       break
     fi
   done
+
+  ver="${ver:1}"
+
+  IFS='.' read -r -a $2 <<< "$ver"
```

Comment:
I do not really see the point on progressive regex parsing if we end up doing a split by `.` in the end anyway

Reply:
We get the best of both worlds like so: support for Bash 3, no dependency on Perl and a much faster parsing.